### PR TITLE
cdap-20173 fix replication UI loading state on assessment error

### PR DESCRIPTION
**Pull request by GnsP**
# [cdap-20173](https://cdap.atlassian.net/browse/CDAP-20173) fix replication UI loading state on assessment error

[fix] replication UI gets blocked in case of backend error on table assessment

## Description
Fixes the following:

```
Steps to reproduce:-
1. Start creating a SQL server replication pipeline
2. Reach Select table and transformations
3. Click on a table which has column of type datetimeoffset
4. Observe that server returns an error response, but UI continues to show in progress screen indefinitely.
```

## PR Type
- [x] Bug Fix
- [ ] Feature
- [ ] Build Fix
- [ ] Testing
- [ ] General Improvement
- [ ] Cherry Pick

## Links
Jira: [20173](https://cdap.atlassian.net/browse/CDAP-20173)

## Test Plan
Manual

## Screenshots
Screenshot 2023-02-08 at 12 12 18 PM

### CODE REVIEW

**Comment by seanfeld** [comment]
Please put something in the title so we have a little bit more context besides 'fix cdap' so we have a little bit of context quickly - like 'cdap-20173 assessment error' or something like that

File: app/cdap/text/text-en.yaml [hunk 1]
```diff
@@ -2870,6 +2870,8 @@ features:
       Content:
         SelectColumns:
           primaryKeyDescription: Must include primary key
+        SelectTables:
+          genericErrorMessage: "Something went wrong!"
```
Comment (seanfeld) [inline app/cdap/text/text-en.yaml]: remove the exclamation point

File: app/cdap/components/Replicator/constants.tsx [hunk 2]
```diff
@@ -23,3 +23,8 @@ export const PROGRAM_INFO = {
   programType: 'workers',
   programId: 'DeltaWorker',
 };
+
+export enum ErrorType {
+  generic = 'generic',
```
Comment (seanfeld) [inline app/cdap/components/Replicator/constants.tsx]: is this generic type being used?

File: app/cdap/text/text-en.yaml (hunk 1 above)
Comment (GnsP) [reply app/cdap/text/text-en.yaml]: done.

File: app/cdap/components/Replicator/constants.tsx (hunk 2 above)
Comment (GnsP) [reply app/cdap/components/Replicator/constants.tsx]: nope, it's not used as of now. Removed it.

**Comment by sumengwang** [comment]
also please change the PR title as well

File: app/cdap/components/Replicator/Create/Content/SelectColumnsWithTransforms/index.tsx [hunk 3]
```diff
@@ -164,7 +165,10 @@ const SelectColumnsView = (props: ISelectColumnsProps) => {
         });
       },
       error: (err) => {
+        dispatch({ type: 'setLoading', payload: false });
         dispatch({ type: 'setError', payload: err });
```
Comment (sumengwang) [inline app/cdap/components/Replicator/Create/Content/SelectColumnsWithTransforms/index.tsx:169]: are there cases we don't want to set loading to false when we set Error? If there's no such case we should just set loading to false in the setError dispatch and use 1 dispatch

File: app/cdap/components/Replicator/Create/Content/SelectTables/index.tsx [hunk 4]
```diff
@@ -45,6 +49,8 @@ import {
   ITableInfo,
 } from 'components/Replicator/types';
 
+const PREFIX = `features.Replication.Create.Content.SelectTables`;
```
Comment (sumengwang) [inline app/cdap/components/Replicator/Create/Content/SelectTables/index.tsx:52]: while you are working on this file, can you also change other hardcoded string into i18n?

**Comment by GnsP** [comment]
> also please change the PR title as well

done

File: app/cdap/components/Replicator/Create/Content/SelectColumnsWithTransforms/index.tsx (hunk 3 above)
Comment (GnsP) [reply app/cdap/components/Replicator/Create/Content/SelectColumnsWithTransforms/index.tsx:169]: There are no other cases where we are using the `setError` dispatch (on this store).
But,I think, it's better to keep these as 2 dispatches, as these are 2 independent concerns. Currently there are no cases where we don't want to set loading to false when we set Error; but that does not mean there can not be any cases when we should not set loading to false when we set error.

File: app/cdap/components/Replicator/Create/Content/SelectTables/index.tsx (hunk 4 above)
Comment (GnsP) [reply app/cdap/components/Replicator/Create/Content/SelectTables/index.tsx:52]: done

**Review by sumengwang — APPROVED** [review]

**Review by seanfeld — APPROVED** [review]

**Pull request merged**